lxd/storage.go

```diff
@@ -187,24 +200,20 @@ func storageForFilename(d *Daemon, filename string) (storage, error) {
 	config := make(map[string]interface{})
 	storageType := storageTypeDir
 
-	filesystem, err := filesystemDetect(filename)
-	if err != nil {
-		return nil, fmt.Errorf("couldn't detect filesystem for '%s': %v", filename, err)
-	}
-
-	lvLinkPath := filename + ".lv"
-	if shared.PathExists(lvLinkPath) {
+	if shared.PathExists(filename + ".lv") {
 		storageType = storageTypeLvm
-		lvPath, err := os.Readlink(lvLinkPath)
+		lvPath, err := os.Readlink(filename + ".lv")
 		if err != nil {
-			return nil, fmt.Errorf("couldn't read link dest for '%s': %v", lvLinkPath, err)
+			return nil, fmt.Errorf("couldn't read link dest for '%s': %v", filename+".lv", err)
 		}
 		vgname := filepath.Base(filepath.Dir(lvPath))
 		config["vgName"] = vgname
-
-	} else if filesystem == "btrfs" {
+	} else if shared.PathExists(filename + ".btrfs") {
 		storageType = storageTypeBtrfs
+	} else if shared.PathExists(filename + ".zfs") {
+		storageType = storageTypeZfs
```

Comment:
Can you get the zfs pool from the path here?
It looks like this is missing a line that sets config["zfsPool"], because it's checked in Init(), but not set anywhere in this patch.